### PR TITLE
Tool fixes.

### PR DESCRIPTION
**Pull request by geeknoid**
- The proto doc generator now only emits the last component of
long type names when generating tables with the proto fields.
Outputting the full dotted names could lead to terrible formatting
as can be seen currently on the MeshConfig reference page.

- Prevent the license-linter from treating the current repo's code as a
dependency, which led to incorrect license errors.

### CODE REVIEW

File: cmd/license-lint/collector.go [hunk 1]
```diff
@@ -160,14 +168,19 @@ var supportedLicenseFilenames = map[string]struct{}{
...
-	err := filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+	err := filepath.Walk(basepath, func(path string, info os.FileInfo, err error) error {
 		if info == nil {
 			return fmt.Errorf("unable to get information on %s: %v", path, err)
 		}
 
-		if !info.IsDir() {
+		if info.IsDir() {
+			if path == filepath.Join(basepath, "licenses") {
+				// don't recurse into this since it holds downstream licenses
```
Comment (sdake) [inline cmd/license-lint/collector.go]: you mean upstream licenses here for the comment?

File: cmd/protoc-gen-docs/htmlGenerator.go [hunk 2]
```diff
@@ -762,7 +762,7 @@ func (g *htmlGenerator) generateComment(loc protomodel.LocationDescriptor, name
 				typeName := match[end+2 : len(match)-1]
 
 				if o, ok := g.model.AllDescByName["."+typeName]; ok {
-					return g.linkify(o, linkName)
+					return g.linkify(o, linkName, false)
```
Comment (sdake) [inline cmd/protoc-gen-docs/htmlGenerator.go]: no idea what the false signifies.

**Comment by geeknoid** [comment]
The HTML change doesn't relate to the license linting change, so no, I'm not producing an HTML rendition of the licenses :-)

File: cmd/license-lint/collector.go (hunk 1 above)
Comment (geeknoid) [reply cmd/license-lint/collector.go]: Yes, fixed.

File: cmd/protoc-gen-docs/htmlGenerator.go (hunk 2 above)
Comment (geeknoid) [reply cmd/protoc-gen-docs/htmlGenerator.go]: Whether to render a type as fully dotted (google.protobuf.Duration) or trim it down (Duration)

**Pull request merged**